oc-game-swift/Players/Player.swift

```diff
@@ -36,35 +42,166 @@ class Player {
+            // Print the message
+            print("-----------------------------------"
+                + "\nChoose a target from the team of \(player.name)"
+                + "\n-----------------------------------"
+                + "\n1. \(player.heroTeam[0].description())"
+                + "\n2. \(player.heroTeam[1].description())"
+                + "\n3. \(player.heroTeam[2].description())")
+            
+            // Choose a hero
+            if let targetHero = chooseHero(player) {
+                heroChosen = true
+                return targetHero
+            }
+        }
+    }
+    
+    /// Player have to choose a hero to heal with the healer
+    /// - return: Hero?, The selected hero
+    func chooseHeroToHeal() -> Hero? {
+        // Hero is choose ?
+        var heroChosen = false
+        // While not
+        while heroChosen == false {
+            // Print the message
+            print("-----------------------------------"
+                + "\n\(self.name), choose the hero to heal"
+                + "\n-----------------------------------"
+                + "\n1. \(self.heroTeam[0].description())"
+                + "\n2. \(self.heroTeam[1].description())"
+                + "\n3. \(self.heroTeam[2].description())")
+            
+            // Choose a hero
+            if let heroToHeal = chooseHero(self) {
+                heroChosen = true
+                return heroToHeal
+            }
+        }
+    }
+    
+    /// Choosing a hero from heroTeam array
+    /// - player: Player, The one who chose
+    /// - return: Hero?, The selected hero
+    private func chooseHero(_ player: Player) -> Hero? {
+        // Read the choice
+        if let choice = readLine() {
+            // If choice is not empty
+            guard !choice.isEmpty else {
+                print("you have to choose a hero")
+                return nil
+            }
+            // Try to cast the String to Int
+            guard let numberChoice = Int(choice) else {
+                print("I dont understand your choice...")
+                return nil
+            }
+            // If numberChoice is not out of range
+            guard numberChoice <= maxHeroes, numberChoice > 0 else {
+                print("You only have \(maxHeroes) heroes")
+                return nil
+            }
+            // Check the status of the selected heroe
+            guard heroIsAlive(player.heroTeam[numberChoice - 1]) else {
+                print("you have to choose a hero")
+                return nil
+            }
+            // All is good
+            return player.heroTeam[numberChoice - 1]
+        }
+        // If the readLine fails
+        print("I can't ask for your choice...")
+        return nil
+    }
+    
+    /// Check the life of hero chosen
+    /// - hero: Hero. Hero chosen
+    /// - return: Bool. If is dead or not
+    private func heroIsAlive(_ hero: Hero) -> Bool {
+        if hero.isDead == false {
```

Comment:
you should factorize these conditions with only one line

Reply:
Done